datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -206,6 +207,36 @@ def list_nodes(
     ]
 
 
+@router.get("/nodes/details/", response_model=List[NodeMinimumDetail])
+def list_nodes_with_details(
+    node_type: Optional[NodeType] = None,
+    *,
+    session: Session = Depends(get_session),
+    current_user: Optional[User] = Depends(get_current_user),
+    validate_access: access.ValidateAccessFn = Depends(  # pylint: disable=W0621
+        validate_access,
+    ),
+) -> List[NodeMinimumDetail]:
+    """
+    List the available nodes.
+    """
+    nodes_query = select(Node).where(
+        Node.current_version == NodeRevision.version,
+        Node.name == NodeRevision.name,
+        Node.type == node_type if node_type else True,
+        is_(Node.deactivated_at, None),
+    )
+    nodes = session.exec(nodes_query).all()
```

Comment:
Good point, I'll update the query to just select those 4 fields.

Reply:
@agorajek that's a good idea to add a very high limit here to be safe. Does that mean the query should have an order by as well? Maybe order by the updated at column asc?

Reply:
No, I don't think you want to spend time sorting these nodes here. I think the limit should be only to switch on an error label on the log.

Reply:
> ...Also it might be better to explicitly pick the columns we want to query on `Node` rather than pulling them all and then relying on `NodeMinimumDetail` to winnow, since it looks like the search only relies on four fields?

@shangyian just an FYI, I'm going to merge this PR and then make this change in a follow-up PR. I had to make some small changes to the authorization logic so that it can accept this minimal set of values (instead of entire Node/NodeRevision instances). I have it mostly done but branched it off so that I can open it separately for review without all of the noise from the main purpose of this PR. cc: @CircArgs